app/src/main/java/com/iic/shopingo/ui/request_flow/RequestStateActivity.java

```diff
@@ -0,0 +1,45 @@
+package com.iic.shopingo.ui.request_flow;
+
+import android.os.Bundle;
+import android.support.v7.app.ActionBarActivity;
+import android.view.View;
+import com.iic.shopingo.R;
+
+public class RequestStateActivity extends ActionBarActivity {
+
+  public static final String REQUEST_STATE_EXTRA_KEY = "request_state";
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    String state = getIntent().getStringExtra(REQUEST_STATE_EXTRA_KEY);
+    int layout;
+    switch (state) {
+      case "approved":
```

Comment:
instead of strings use a Java enum and then instead of getStringExtra use getIntExtra